cryptomator: Cannot mount vault as FUSE volume

### Describe the bug
In the vault options, vaults can only be mounted as `WebDAV (HTTP Address)`.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Launch cryptomator
2. Go to a vault options
3. In the `Virtual Drive` tab, the `Volume Type` combo box does not contain a FUSE item

### Expected behavior
Vaults can be mounted as FUSE volumes

### Additional context
In the console output, cryptomator seems to have access to the fuse3 libs in its PATH:
```
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - java.library.path: /nix/store/gplxx8rm08abic60i4iib99pcabbdgq1-fuse-3.11.0/lib:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
```

Full console output:
```
14:19:36.290 [main] INFO  o.c.common.settings.SettingsProvider - Settings loaded from /home/nicomem/.config/Cryptomator/settings.json
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - user.home: /home/nicomem
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - java.library.path: /nix/store/gplxx8rm08abic60i4iib99pcabbdgq1-fuse-3.11.0/lib:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - user.language: en
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - user.region: null
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - logback.configurationFile: null
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.settingsPath: ~/.config/Cryptomator/settings.json
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.ipcSocketPath: ~/.config/Cryptomator/ipc.socket
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.integrationsWin.keychainPaths: null
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.logDir: ~/.local/share/Cryptomator/logs
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.loopbackAlias: null
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.pluginDir: ~/.local/share/Cryptomator/plugins
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.mountPointsDir: ~/.local/share/Cryptomator/mnt
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.minPwLength: null
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.appVersion: 1.9.4
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.buildNumber: nix
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.showTrayIcon: false
14:19:36.332 [main] INFO  org.cryptomator.common.Environment - cryptomator.p12Path: null
14:19:36.332 [main] INFO  o.cryptomator.launcher.Cryptomator - Starting Cryptomator 1.9.4 on Linux 6.5.4 (amd64)
14:19:36.342 [main] INFO  org.cryptomator.ipc.Server - Spawning IPC server listening on socket /home/nicomem/.config/Cryptomator/ipc.socket
14:19:36.483 [JavaFX Application Thread] INFO  o.cryptomator.launcher.Cryptomator - JavaFX runtime started after 302ms
```

### Notify maintainers
@bachp 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.5.4, NixOS, 23.11 (Tapir), 23.11.20230919.5ba549e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos"`
 - channels(nicomem): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

thank you for your response!

i hope i am not rude to ask for your help in achieving this myself.

i put the following in my `home.nix`:

```nix
overlays = [
  (final: prev: {
    cryptomator = prev.cryptomator.overrideAttrs (old: {
      src = prev.fetchFromGitHub {
        owner = "cryptomator";
        repo = "cryptomator";
        rev = "1.11.0-beta2";
        hash = "sha256-QYVvu1BrOGoz+gOGXGt3Y08H8ufnHhQDDp1H/MsDC/k=";
      };
    });
  })
];
```

and i also did the `jdk21` overlay.

i get the following error when building:

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/1c6xig7pgprxfannqhzrnrh76naz6d87-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< org.cryptomator:cryptomator >---------------------
[INFO] Building Cryptomator Desktop App 1.11.0
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for org.codehaus.mojo:license-maven-plugin:jar:2.2.0 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.223 s
[INFO] Finished at: 2023-11-01T15:27:35Z
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.codehaus.mojo:license-maven-plugin:2.2.0 or one of its dependencies could not be resolved: Cannot access central (https://repo.maven.apache.org/maven2) in offline mode and the artifact org.codehaus.mojo:license-maven-plugin:jar:2.2.0 has not been downloaded from it before. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
```

haven't been able to successfully find a solution via search engines...

--- a/pkgs/tools/security/cryptomator/default.nix
+++ b/pkgs/tools/security/cryptomator/default.nix
@@ -13,17 +13,17 @@ in
 assert stdenv.isLinux; # better than `called with unexpected argument 'enableJavaFX'`
 mavenJdk.buildMavenPackage rec {
   pname = "cryptomator";
-  version = "1.10.1";
+  version = "1.11.0";
 
   src = fetchFromGitHub {
     owner = "cryptomator";
     repo = "cryptomator";
     rev = version;
-    hash = "sha256-xhj7RUurBRq9ZIDAlcq7KyYGnLqc+vTjaf2VMNStpVQ";
+    hash = "sha256-NMNlDEUpwKUywzhXhxlNX7NiE+6wOov2Yt8nTfbKTNI=";
   };
 
   mvnParameters = "-Dmaven.test.skip=true";
-  mvnHash = "sha256-XAIwKn8wMqILMQbg9wM4kHAaRSGWQaBx9AXQyJuUO5k=";
+  mvnHash = "sha256-jIHMUj7ZQFu4XAvWUywj4f0PbmLHGtU5VRG0ZuKm3mA=";
 
   preBuild = ''
     VERSION=${version}
@@ -42,11 +42,13 @@ mavenJdk.buildMavenPackage rec {
       --add-flags "--enable-preview" \
       --add-flags "--class-path '$out/share/cryptomator/libs/*'" \
       --add-flags "--module-path '$out/share/cryptomator/mods'" \
-      --add-flags "-Dcryptomator.logDir='~/.local/share/Cryptomator/logs'" \
-      --add-flags "-Dcryptomator.pluginDir='~/.local/share/Cryptomator/plugins'" \
-      --add-flags "-Dcryptomator.settingsPath='~/.config/Cryptomator/settings.json'" \
-      --add-flags "-Dcryptomator.ipcSocketPath='~/.config/Cryptomator/ipc.socket'" \
-      --add-flags "-Dcryptomator.mountPointsDir='~/.local/share/Cryptomator/mnt'" \
+      --add-flags "-Dfile.encoding='utf-8'" \
+      --add-flags "-Dcryptomator.logDir='@{userhome}/.local/share/Cryptomator/logs'" \
+      --add-flags "-Dcryptomator.pluginDir='@{userhome}/.local/share/Cryptomator/plugins'" \
+      --add-flags "-Dcryptomator.settingsPath='@{userhome}/.config/Cryptomator/settings.json'" \
+      --add-flags "-Dcryptomator.p12Path='@{userhome}/.config/Cryptomator/key.p12'" \
+      --add-flags "-Dcryptomator.ipcSocketPath='@{userhome}/.config/Cryptomator/ipc.socket'" \
+      --add-flags "-Dcryptomator.mountPointsDir='@{userhome}/.local/share/Cryptomator/mnt'" \
       --add-flags "-Dcryptomator.showTrayIcon=false" \
       --add-flags "-Dcryptomator.buildNumber='nix'" \
       --add-flags "-Dcryptomator.appVersion='${version}'" \

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -27710,7 +27710,7 @@ with pkgs;
   criu = callPackage ../os-specific/linux/criu { };
 
   cryptomator = callPackage ../tools/security/cryptomator {
-    jdk = jdk20.override { enableJavaFX = true; };
+    jdk = jdk21.override { enableJavaFX = true; };
   };
 
   cryptsetup = callPackage ../os-specific/linux/cryptsetup { };

